diff --git a/cmd/root.go b/cmd/root.go
index 61a163e..da3325d 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -83,6 +83,7 @@ func init() {
 	rootCmd.PersistentFlags().BoolP("no-color", "", false, "turn off color for verbose output")
 	rootCmd.PersistentFlags().Int("max-target-megabytes", 0, "files larger than this will be skipped")
 	rootCmd.PersistentFlags().BoolP("ignore-gitleaks-allow", "", false, "ignore gitleaks:allow comments")
+	rootCmd.PersistentFlags().Bool("no-source-ignore", false, "do not use a source-controlled .gitleaksignore file as scanner policy")
 	rootCmd.PersistentFlags().Uint("redact", 0, "redact secrets from logs and stdout. To redact only parts of the secret just apply a percent value from 0..100. For example --redact=20 (default 100%)")
 	rootCmd.Flag("redact").NoOptDefVal = "100"
 	rootCmd.PersistentFlags().Bool("no-banner", false, "suppress banner")
@@ -297,26 +298,33 @@ func Detector(cmd *cobra.Command, cfg config.Config, source string) *detect.Dete
 		logging.Fatal().Err(err).Send()
 	}
 
-	gitleaksIgnorePath, err := cmd.Flags().GetString("gitleaks-ignore-path")
+	noSourceIgnore, err := cmd.Flags().GetBool("no-source-ignore")
 	if err != nil {
-		logging.Fatal().Err(err).Msg("could not get .gitleaksignore path")
+		logging.Fatal().Err(err).Msg("could not get no-source-ignore")
 	}
 
-	if fileExists(gitleaksIgnorePath) {
-		if err = detector.AddGitleaksIgnore(gitleaksIgnorePath); err != nil {
-			logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+	if !noSourceIgnore {
+		gitleaksIgnorePath, err := cmd.Flags().GetString("gitleaks-ignore-path")
+		if err != nil {
+			logging.Fatal().Err(err).Msg("could not get .gitleaksignore path")
 		}
-	}
 
-	if fileExists(filepath.Join(gitleaksIgnorePath, ".gitleaksignore")) {
-		if err = detector.AddGitleaksIgnore(filepath.Join(gitleaksIgnorePath, ".gitleaksignore")); err != nil {
-			logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+		if fileExists(gitleaksIgnorePath) {
+			if err = detector.AddGitleaksIgnore(gitleaksIgnorePath); err != nil {
+				logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+			}
 		}
-	}
 
-	if fileExists(filepath.Join(source, ".gitleaksignore")) {
-		if err = detector.AddGitleaksIgnore(filepath.Join(source, ".gitleaksignore")); err != nil {
-			logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+		if fileExists(filepath.Join(gitleaksIgnorePath, ".gitleaksignore")) {
+			if err = detector.AddGitleaksIgnore(filepath.Join(gitleaksIgnorePath, ".gitleaksignore")); err != nil {
+				logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+			}
+		}
+
+		if fileExists(filepath.Join(source, ".gitleaksignore")) {
+			if err = detector.AddGitleaksIgnore(filepath.Join(source, ".gitleaksignore")); err != nil {
+				logging.Fatal().Err(err).Msg("could not call AddGitleaksIgnore")
+			}
 		}
 	}
 
